runtime/sys.c

```diff
@@ -439,7 +439,11 @@ CAMLprim value caml_sys_system_command(value command)
   }
   buf = caml_stat_strdup_to_os(String_val(command));
   caml_enter_blocking_section ();
-  status = system_os(buf);
+  #if HAS_SYSTEM
+    status = system_os(buf);
+  #else
+    caml_invalid_argument("Sys.command not implemented");
+  #endif /* HAS_SYSTEM */
   caml_leave_blocking_section ();
```

Comment:
Nice PR! Just a quick comment: caml_invalid_argument here must occur after the `caml_leave_blocking_section` (after which point you have the runtime lock, and so can raise an exception).

Reply:
Nice catch, addressed by calling `caml_enter_blocking_section` and `caml_leave_blocking_section` only when `HAS_SYSTEM`